mono/utils/mono-util.h

```diff
@@ -0,0 +1,24 @@
+/**
+ * \file
+ */
+
+#ifndef __UTILS_MONO_UTIL_H__
```

Comment:
1. I think you need to add this file to `mono/utils/Makefile.am` to `monoutils_sources`.
2. Our usual prefix is `__MONO_`, so: `__MONO_UTILS_MONO_UTIL_H`

Reply:
Do we really need a new file for this one definition ?

Reply:
I agree it is overkill but I do like "util.h", it is where I used to put almost everything, everything with a shred of genericity. I can put it elsewhere. I think I stopped using it, or can stop using it anyway. But I do like this MONO_CONST_CAST, if we are ever to move to C++.. If not, well, cast all over the place..

Reply:
I thought I was following convention on the ifdef guard but I'll check again. Makefile is only for installed stuff, and we want to install the minimum, right?

Reply:
There are indeed multiple conventions.
I followed one of them.

jays-MacBook-Pro:mono3 jay$ grep ifndef mono/utils/*.h | grep " _MONO"  |wc -l
       9
jays-MacBook-Pro:mono3 jay$ grep ifndef mono/utils/*.h | grep " __MONO"  |wc -l
      58
jays-MacBook-Pro:mono3 jay$ grep ifndef mono/utils/*.h | grep " MONO"  |wc -l
       7

jays-MacBook-Pro:mono3 jay$ grep ifndef mono/utils/*.h | grep "UTILS" |wc -l
      28
jays-MacBook-Pro:mono3 jay$ grep ifndef mono/utils/*.h  |wc -l
     192
     
Just a few, with inconsistencies:

mono/utils/bsearch.h:#ifndef __MONO_BSEARCH_H__
mono/utils/gc_wrapper.h:#ifndef __MONO_OS_GC_WRAPPER_H__
mono/utils/hazard-pointer.h:#ifndef __MONO_HAZARD_POINTER_H__
mono/utils/json.h:#ifndef __MONO_UTILS_JSON_H__
mono/utils/lock-free-alloc.h:#ifndef __MONO_LOCKFREEALLOC_H__
mono/utils/lock-free-array-queue.h:#ifndef __MONO_LOCK_FREE_ARRAY_QUEUE_H__
mono/utils/lock-free-queue.h:#ifndef __MONO_LOCKFREEQUEUE_H__
mono/utils/mach-support.h:#ifndef __MONO_MACH_SUPPORT_H__
mono/utils/memfuncs.h:#ifndef __MONO_UTILS_MEMFUNCS_H__
mono/utils/mono-codeman.h:#ifndef __MONO_CODEMAN_H__
mono/utils/mono-compiler.h:#ifndef __UTILS_MONO_COMPILER_H__
mono/utils/mono-conc-hashtable.h:#ifndef __MONO_CONCURRENT_HASHTABLE_H__
mono/utils/mono-context.h:#ifndef __MONO_MONO_CONTEXT_H__
mono/utils/mono-coop-mutex.h:#ifndef __MONO_COOP_MUTEX_H__
mono/utils/mono-coop-semaphore.h:#ifndef __MONO_COOP_SEMAPHORE_H__
mono/utils/mono-counters.h:#ifndef __MONO_COUNTERS_H__
mono/utils/mono-digest.h:#ifndef __MONO_DIGEST_H__
mono/utils/mono-dl-fallback.h:#ifndef __MONO_UTILS_DL_FALLBACK_H__
mono/utils/mono-dl-windows-internals.h:#ifndef __MONO_UTILS_DL_WINDOWS_H__
mono/utils/mono-dl.h:#ifndef __MONO_UTILS_DL_H__
mono/utils/mono-dtrace.h:#ifndef        _MONO_DTRACE_H
mono/utils/mono-embed.h:#ifndef __MONO_EMBED_H__
mono/utils/mono-error-internals.h:#ifndef __MONO_ERROR_INTERNALS_H__
mono/utils/mono-error.h:#ifndef __MONO_ERROR_H__
mono/utils/mono-hwcap-sparc.h:#ifndef __MONO_UTILS_HWCAP_SPARC_H__
mono/utils/mono-hwcap.h:#ifndef __MONO_UTILS_HWCAP_H__
mono/utils/mono-internal-hash.h:#ifndef __MONO_UTILS_MONO_INTERNAL_HASH__
mono/utils/mono-io-portability.h:#ifndef __MONO_IO_PORTABILITY_H
mono/utils/mono-lazy-init.h:#ifndef __MONO_LAZY_INIT_H__
mono/utils/mono-linked-list-set.h:#ifndef __MONO_SPLIT_ORDERED_LIST_H__
mono/utils/mono-logger-internals.h:#ifndef __MONO_LOGGER_INTERNAL_H__
mono/utils/mono-logger.h:#ifndef __MONO_LOGGER_H__
mono/utils/mono-machine.h:#ifndef __MONO_MONO_MACHINE_H__
mono/utils/mono-math.h:#ifndef __MONO_SIGNBIT_H__
mono/utils/mono-membar.h:#ifndef _MONO_UTILS_MONO_MEMBAR_H_
mono/utils/mono-memory-model.h:#ifndef _MONO_UTILS_MONO_MEMMODEL_H_
mono/utils/mono-mmap-internals.h:#ifndef __MONO_UTILS_MMAP_INTERNAL_H__
mono/utils/mono-mmap-windows-internals.h:#ifndef __MONO_UTILS_MMAP_WINDOWS_H__
mono/utils/mono-mmap.h:#ifndef __MONO_UTILS_MMAP_H__
mono/utils/mono-networkinterfaces.h:#ifndef __MONO_NETWORK_INTERFACES_H__
mono/utils/mono-once.h:#ifndef __MONO_ONCE_H__

Reply:
Um, maybe don't trust GitHub gui, underscores might be bold?

Reply:
ok I removed the include, though the file remains at least for now.
https://github.com/jaykrell/mono/commit/94abf76ae01f62c147468d1252e2398358910f1b